src/is_intersection_empty.jl

```diff
@@ -595,3 +596,83 @@ function is_intersection_empty(set::AbstractPolytope{N},
                               )::Union{Bool, Tuple{Bool, Vector{N}}} where {N<:Real}
     return is_intersection_empty(point, set, witness)
 end
+
+# --- Hyperplane ---
+
+"""
+    is_intersection_empty(X::LazySet{N},
+                          hp::Union{Hyperplane{N}, Line{N}},
+                          [witness]::Bool=false
+                         )::Union{Bool, Tuple{Bool,Vector{N}}} where N<:Real
+
+Check whether a compact set an a hyperplane do not intersect, and otherwise
+optionally compute a witness.
+
+### Input
+
+- `X`       -- compact set
+- `hp`      -- hyperplane
+- `witness` -- (optional, default: `false`) compute a witness if activated
+
+### Output
+
+* If `witness` option is deactivated: `true` iff ``P ∩ Q = ∅``
+* If `witness` option is activated:
+  * `(true, [])` iff ``P ∩ Q = ∅``
+  * `(false, v)` iff ``P ∩ Q ≠ ∅`` and ``v ∈ P ∩ Q``
+
+### Notes
+
+We assume that `X` is compact.
+Otherwise, the support vector queries may fail.
+
+### Algorithm
+
+A compact convex set intersects with a hyperplane iff the support function in
+the negative resp. positive direction of the hyperplane's normal vector ``a`` is
+to the left resp. right of the hyperplane's constraint ``b``:
+
+```math
+-ρ(-a) ≤ b ≤ ρ(a)
+```
+
+For witness generation, we compute a line connecting the support vectors to the
+left and right, and then take the intersection of the line with the hyperplane.
+We follow
+[this algorithm](https://en.wikipedia.org/wiki/Line%E2%80%93plane_intersection#Algebraic_form)
+for the line-hyperplane intersection.
```

Comment:
Nice idea! 

I was thinking how can we use it outside this function but then realized we don't have "hyperlines". See #652.